src/backend/vulkan/src/window.rs

```diff
@@ -308,11 +318,15 @@ impl w::Surface<Backend> for Surface {
     fn supported_formats(&self, physical_device: &PhysicalDevice) -> Option<Vec<Format>> {
         // Swapchain formats
         let raw_formats = unsafe {
-            self.raw
+            match self.raw
                 .functor
-                .get_physical_device_surface_formats(physical_device.handle, self.raw.handle)
-        }
-        .expect("Unable to query surface formats");
+                .get_physical_device_surface_formats(physical_device.handle, self.raw.handle) {
+                    Ok(formats) => formats,
+                    #[cfg(target_os = "android")]
+                    Err(vk::Result::ERROR_SURFACE_LOST_KHR) => return None,
```

Comment:
This is a bit more tricky, unfortunately. `None` result means that all formats are supported. Is this really what we want to report if the surface is lost? I think returning `Some(Vec::new())` would be a little better